phase1/azure/do

```diff
@@ -67,13 +67,29 @@ check_auth() {
   fi
 }
 
+check_storage_account_name() {
+  storage_account_name="$(jq -r '.phase1.azure.storage_account_name' ${configjson})"
+  if [[ ! -z "${storage_account_name}" ]]; then
+    # everything is fine, we assume user knows what they are doing
+    return 0
+  fi
+  set +e; storage_account_name="$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 21 | head -n 1)k8s"; set -e
```

Comment:
Crap. I had originally made it be 'k8s{RANDOM}' but then remember what I know about the storage shard hashing function. You're right, I'll try to figure out something that preserves "max" entropy but ensures a letter.

And good catch, thanks.